python/python/tests/test_pubsub.py

```diff
@@ -0,0 +1,486 @@
+                        if channel_and_pattern not in self.messages_to_receive:
+                            self.messages_to_receive[channel_and_pattern] = {}
+                        for message, cnt in unsharded_channel_messages.items():
+                            if message not in self.messages_to_receive[channel_and_pattern]:
+                                self.messages_to_receive[channel_and_pattern][message] = 0
+                            self.messages_to_receive[channel_and_pattern][message] += cnt
+
+                    # should receive by glob pattern?
+                    for glob_pattern in self.subsciptions[CoreCommands.ChannelModes.Pattern]:
+                        if fnmatch.filter(names=[unsharded_channel], pat=glob_pattern):
+                            channel_and_pattern = unsharded_channel, glob_pattern
+                            for message, cnt in unsharded_channel_messages.items():
+                                if message not in self.messages_to_receive[channel_and_pattern]:
+                                    self.messages_to_receive[channel_and_pattern][message] = 0
+                                self.messages_to_receive[channel_and_pattern][message] += cnt
+
+        # TODO: Allow async fanning out?
+        def sched_publish_messages(self) -> List[Awaitable[None]]:
+            # coroutines = []
+            # for sharded_channel, sharded_messages_to_publish in self.sharded_messages_to_publish:
+            #     for message in sharded_messages_to_publish:
+            #         coroutines += self.redis_client.publish(message=message, channels={sharded_channel}, sharded=True)
+
+            # for unsharded_channel, unsharded_messages_to_publish in self.unsharded_messages_to_publish:
+            #     for message in unsharded_messages_to_publish:
+            #         coroutines += self.redis_client.publish(message=message, channels={unsharded_channel}, sharded=False)
+
+            # return coroutines
+            pass
+        
+        async def publish_messages(self) -> List[Awaitable[None]]:
+            for sharded_channel, sharded_messages_to_publish in self.sharded_messages_to_publish:
+                for message, cnt in sharded_messages_to_publish:
+                    for i in range(cnt):
+                        await self.redis_client.publish(message=message, channels={sharded_channel}, sharded=True)
+
+            for unsharded_channel, unsharded_messages_to_publish in self.unsharded_messages_to_publish:
+                for message, cnt in unsharded_messages_to_publish:
+                    for i in range(cnt):
+                        await self.redis_client.publish(message=message, channels={unsharded_channel}, sharded=False)
+
+
+    @staticmethod
+    async def publish_all_messages(all_clients: List[TestPubSub.PubSubClient]):
+        coroutines = []
+        for client in all_clients:
+            coroutines += client.sched_publish_messages()
+        asyncio.wait(coroutines)
```

Comment:
I think https://docs.python.org/3/library/asyncio-task.html#asyncio.gather is what you need here

Reply:
removed for now, done